### DISCUSSION TITLE
Send data from Arduino to Tasmota based ESP32

### DISCUSSION THREAD

**Discussion by Sumya99** [arendst/Tasmota#15622] [Q&A] [answered]
Hello,
I want to send string or a byte data to Tasmota based ESP32 and it shows it as a sensor value on the web server. Is it possible using I2c or SPI protocol?

Thanks in advance

**Comment by barbudor** [comment, marked as answer]
You can use serial with TasmotaClient
https://tasmota.github.io/docs/TasmotaClient/

Going I2c is also possible and take advantage of Berry language on ESP32 to write a driver. Tasmota would act as a i2c controller and arduino as an i2c device.

You can also use serial without TasmotaClient and a driver in Berry

**Comment by s-hadinger** [comment]
I'm not sure to understand what you want to do, but I'm almost certain you can do this with Berry scripting. 

https://tasmota.github.io/docs/Berry/

**Comment by lastmetroid** [comment]
I am also interessted in doing this. I am sure there is a out of the box finished tutorial or project for this already, this is so simple and basic, there needs to be a project for it already to exist.

All I want is to:

Create a sensors json on a Tasmota wifi device like

SENSORS {

"sensor1": "10",
"sensor2": "0",
"sensor3": "1"

}

And send the data via serial to it (echo "10" "sensor1" > serialdevice or something like that), and then make it published on MQTT.

**Reply by sfromis** [reply]
Since this thread is titled with ESP32, using the Berry language makes such manoeuvres reasonably straightforward to code, as there is good support for both JSON, serial and MQTT, along with coding sensor drivers.
